test/Serilog.Tests/MethodOverloadConventionTests.cs

```diff
@@ -12,361 +13,864 @@
+
+                //check for params argument: params object[] propertyValues
+                //params argument currently has to be the last argument, and generic methods don't have params argument
+                if (!isGeneric && (parameters.Length - index) == 1)
+                {
+                    var paramsArrayArg = parameters[index];
+
+                    // params array attribute should never have derived/inherited classes
+                    var paramsAttr = parameters[index].GetCustomAttribute(typeof(ParamArrayAttribute), inherit: false);
+
+                    Assert.NotNull(paramsAttr);
+
+                    Assert.Equal(paramsArrayArg.ParameterType, typeof(object[]));
+
+                    Assert.Equal(paramsArrayArg.Name, "propertyValues");
+                }
+            }
+            catch (XunitException e)
+            {
+                // mark xunit assertion failures
+                e.Data.Add("IsSignatureAssertionFailure", true);
+
+                throw e;
+            }
+        }
+
+        static object InvokeMethod(
+            MethodInfo method,
+            ILogger instance,
+            object[] parameters,
+            Type[] typeArgs = null)
+        {
+            if (method.IsStatic)
+            {
+                if (method.IsGenericMethod)
+                    return method.MakeGenericMethod(typeArgs).Invoke(null, parameters);
+                else
+                    return method.Invoke(null, parameters);
+            }
+            else if (method.IsGenericMethod)
+                return method.MakeGenericMethod(typeArgs).Invoke(instance, parameters);
+            else
+                return method.Invoke(instance, parameters);
+        }
+
+        static void TestResults(LogEventLevel level, CollectingSink results)
+        {
+            Assert.Equal(1, results.Events.Count);
+
+            var evt = results.Events.Single();
+
+            Assert.Equal(level, evt.Level);
+        }
+
+        static ILogger GetLogger(Type loggerType)
+        {
+            CollectingSink sink;
+
+            return GetLogger(loggerType, out sink);
+        }
+
+        static ILogger GetLogger(Type loggerType, out CollectingSink sink)
+        {
+            sink = null;
+
+            if (loggerType == typeof(Logger) || loggerType == typeof(ILogger))
+            {
+                sink = new CollectingSink();
+
+                return new LoggerConfiguration()
+                    .MinimumLevel.Verbose()
+                    .WriteTo.Sink(sink)
+                    .CreateLogger();
+            }
+            else if (loggerType == typeof(Log))
+            {
+                sink = new CollectingSink();
+
+                Log.CloseAndFlush();
+
+                Log.Logger = new LoggerConfiguration()
+                    .MinimumLevel.Verbose()
+                    .WriteTo.Sink(sink)
+                    .CreateLogger();
+
+                return null;
+            }
+            else
```

Comment:
Perhaps adding the `if (loggerType == typeof(ILogger) || loggerType == typeof(SilentLogger)` test here, and an `else` branch that throws `ArgumentException`, will help prevent issues creeping in?